components/motor/gpio/encoded.go

```diff
@@ -261,9 +265,9 @@ func (m *EncodedMotor) GoFor(ctx context.Context, rpm, revolutions float64, extr
 
 	positionReached := func(ctx context.Context) (bool, error) {
 		var errs error
-		pos, posErr := m.position(ctx, extra)
+		currentTicks, _, posErr := m.encoder.Position(ctx, encoder.PositionTypeTicks, extra)
 		errs = multierr.Combine(errs, posErr)
-		if (direction == 1 && pos >= goalPos) || (direction == -1 && pos <= goalPos) {
+		if (goalPos-currentTicks)*direction > 0 {
```

Comment:
still not the biggest fan of needing both this check and the check in `makeAdjustments`. But fine with leaving it since this check should go away once we can remove the SetRPM behavior from here